functions/APIs/queries.js

```diff
@@ -0,0 +1,308 @@
+                data["durationText"] = durationText.join(" ");
+                resolve(data);
+            }
+            else {
+                resolve();
+            }
+        })
+        .catch((err) => {
+            return reject(err);
+        });
+    })
+}
+
+//convert time from days, hours, and minutes to minutes
+function convertTime(days, hours, minutes) {
+    return (24 * 60 * days) + (60 * hours) + minutes;
+}
+
+/*
+find food banks that are a specific distance from a farm
+GET /queryFoodBanksByDistance
+URL params: locationId, distance (miles)
+success response: array of food bank objects
+*/
+exports.queryFoodBanksByDistance = (request, response) => {
+    db.collection('foodbanks').get()
+    .then((data) => {
+        let foodbanks = [];
+        data.forEach((doc) => {
+            foodbanks.push(doc.data());
+        });
+
+        Promise.all(foodbanks.map(function(foodbank) {
+            return calculateDistances(foodbank, request);
+        }))
+        .then((results) => {
+            results = results.filter(result => result != null);
+            results = _.orderBy(results, ["distanceValue"]);
+            return response.json(results);
+        })
+        .catch((err) => {
+            console.log(err);
+        });
+    })
+    .catch((err) => {
+        console.log(err);
+        return response.status(500).json({error: err.code});
+    });
+}
+
+/*
+find farms that are a specific distance from a food bank
+GET /queryFarmsByDistance
+URL params: locationId, distance (miles)
+success response: array of farm objects
+*/
+exports.queryFarmsByDistance = (request, response) => {
+    db.collection('farms').get()
+    .then((data) => {
+        let farms = [];
+        data.forEach((doc) => {
+            farms.push(doc.data());
+        });
+
+        Promise.all(farms.map(function(farm) {
+            return calculateDistances(farm, request);
+        }))
+        .then((results) => {
+            results = results.filter(result => result != null);
+            results = _.orderBy(results, ["distanceValue"]);
+            return response.json(results);
+        })
+        .catch((err) => {
+            console.log(err);
+        });
+    })
+    .catch((err) => {
+        console.log(err);
+        return response.status(500).json({error: err.code});
+    });
+}
+
+/*
+find farms with surplus that are a specific distance from a food bank
+GET /querySurplusByDistance
+URL params: locationId, distance (miles)
+success response: array of surplus objects
+*/
+exports.querySurplusByDistance = (request, response) => {
+    db.collection('surplus').get()
+    .then((data) => {
+        let surplus = [];
+        data.forEach((doc) => {
+            surplus.push(doc.data());
+        });
+
+        function queryFarms(data) {
+            return new Promise(function(resolve, reject) {
+                db.doc(`/farms/${data.originFarmId}`).get()
+                .then((doc) => {
+                    let surplus = data;
```

Comment:
pointing this out as well, im not sure which surplus is being expanded on line 157